client/client.go

```diff
@@ -862,11 +862,20 @@ func (c *Client) waitForReplyWithCancel(id wamp.ID, cancelAfter time.Duration, m
 		return nil, fmt.Errorf("not expecting reply for ID: %v", id)
 	}
 
+	// If the context does not have a deadline, then give it the default
+	// response timeout.
+	if _, ok := ctx.Deadline(); !ok {
+		var cancel context.CancelFunc
+		ctx, cancel = context.WithTimeout(ctx, c.responseTimeout)
+		defer cancel()
+	}
+
 	var msg wamp.Message
 	var err error
 	select {
 	case msg = <-wait:
-	case <-time.After(cancelAfter):
+	case <-ctx.Done():
+		log.Print("Caller cancelling call %v (mode=%s)", id, mode)
```

Comment:
This log entry seems a bit confusing. Isn't it possible that the call was cancelled due to the deadline (and nothing the caller did)?